AsyncDisplayKit/ASImageNode.mm

```diff
@@ -410,6 +415,174 @@ - (void)setImageModificationBlock:(asimagenode_modification_block_t)imageModific
   _imageModificationBlock = imageModificationBlock;
 }
 
+
+#if TARGET_OS_TV
+- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
+{
+  [super touchesBegan:touches withEvent:event];
+  isDefaultState = NO;
+  UIView *view = [self getView];
+  CALayer *layer = view.layer;
+
+  CGSize targetShadowOffset = CGSizeMake(0.0, self.bounds.size.height/8);
+  [layer removeAllAnimations];
+  [CATransaction begin];
+  [CATransaction setCompletionBlock:^{
+    layer.shadowOffset = targetShadowOffset;
+  }];
+  
+  CABasicAnimation *shadowOffsetAnimation = [CABasicAnimation animationWithKeyPath:@"shadowOffset"];
+  shadowOffsetAnimation.toValue = [NSValue valueWithCGSize:targetShadowOffset];
+  shadowOffsetAnimation.duration = 0.4;
+  shadowOffsetAnimation.removedOnCompletion = NO;
+  shadowOffsetAnimation.fillMode = kCAFillModeForwards;
+  shadowOffsetAnimation.timingFunction = [CAMediaTimingFunction functionWithName:@"easeOut"];
+  [layer addAnimation:shadowOffsetAnimation forKey:@"shadowOffset"];
+  [CATransaction commit];
+  
+  CABasicAnimation *shadowOpacityAnimation = [CABasicAnimation animationWithKeyPath:@"shadowOpacity"];
+  shadowOpacityAnimation.toValue = [NSNumber numberWithFloat:0.45];
+  shadowOpacityAnimation.duration = 0.4;
+  shadowOpacityAnimation.removedOnCompletion = false;
+  shadowOpacityAnimation.fillMode = kCAFillModeForwards;
+  shadowOpacityAnimation.timingFunction = [CAMediaTimingFunction functionWithName:@"easeOut"];
+  [layer addAnimation:shadowOpacityAnimation forKey:@"shadowOpacityAnimation"];
+  
+  view.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1.25, 1.25);
+  
+  [CATransaction commit];
+}
+
+- (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event {
+  [super touchesMoved:touches withEvent:event];
+
+  if (!isDefaultState) {
+    UIView *view = [self getView];
+
+    UITouch *touch = [touches anyObject];
+    // Get the specific point that was touched
+    // This is quite messy in it's current state so is not ready for production. The reason it is here is for others to contribute and to make it clear what is occuring.
+    // TODO: Clean up, and improve visuals.
+    CGPoint point = [touch locationInView:self.view];
+    float pitch = 0;
+    float yaw = 0;
+    BOOL topHalf = NO;
+    if (point.y > CGRectGetHeight(self.view.frame)) {
+      pitch = 15;
+    } else if (point.y < -CGRectGetHeight(self.view.frame)) {
+      pitch = -15;
+    } else {
+      pitch = (point.y/CGRectGetHeight(self.view.frame))*15;
+    }
+    if (pitch < 0) {
+      topHalf = YES;
+    }
+    
+    if (point.x > CGRectGetWidth(self.view.frame)) {
+      yaw = 10;
```

Comment:
This all is _very_ cool support to have!  We should probably make either ASImageNode+tvOS or find a way to abstract this code out.  Making a folder for tvOS in the project structure and then building +tvOS category files for anything with more than 50 lines of tvOS support seems like a good guideline.